lib/std/zig.zig

```diff
@@ -49,6 +49,10 @@ pub const Loc = struct {
     column: usize,
     /// Does not include the trailing newline.
     source_line: []const u8,
+
+    pub fn eql(a: Loc, b: Loc) bool {
+        return a.line == b.line and a.column == b.column and std.mem.eql(u8, a.source_line, b.source_line);
```

Comment:
The line comparison is harmless but redundant:

```suggestion
        return a.line == b.line and a.column == b.column;
```

Reply:
It's there because I don't check the locations to be in the same file in any other way. I don't think it is possible for it to be false currently but it also barely has any impact since the first two checks will be the only ones done in most cases.